src/Components/Search/NewSearch/Mobile/SearchResultsList.tsx

```diff
@@ -0,0 +1,247 @@
+import {
+  NewSuggestionItem,
+  SuggionItemOptionProps,
+} from "Components/Search/NewSearch/SuggestionItem/NewSuggestionItem"
+import { Flex, Spinner } from "@artsy/palette"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { SearchResultsListQuery } from "__generated__/SearchResultsListQuery.graphql"
+import { useTracking } from "react-tracking"
+import * as DeprecatedSchema from "@artsy/cohesion/dist/DeprecatedSchema"
+import {
+  SearchNodeOption,
+  formatOptions,
+} from "Components/Search/NewSearch/utils/formatOptions"
+import { SeeFullResults } from "Components/Search/NewSearch/Mobile/SearchResultsList/SeeFullResults"
+import { InfiniteScrollSentinel } from "Components/Search/NewSearch/Mobile/SearchResultsList/InfiniteScrollSentinel"
+import { ContentPlaceholder } from "Components/Search/NewSearch/Mobile/SearchResultsList/ContentPlaceholder"
+import { NoResults } from "Components/Search/NewSearch/Mobile/SearchResultsList/NoResults"
+
+interface SearchResultsListProps {
+  relay: RelayPaginationProp
+  viewer: SearchResultsList_viewer$data
+  query: string
+  onClose: () => void
+}
+
+const ENTITIES_PER_SCROLL = 10
+
+const SearchResultsList: FC<SearchResultsListProps> = ({
+  relay,
+  viewer,
+  query,
+  onClose,
+}) => {
+  const tracking = useTracking()
+  const [isLoading, setIsLoading] = useState(false)
+
+  const options = extractNodes(viewer.searchConnection)
+
+  if (!relay.isLoading() && options.length === 0) {
+    return <NoResults query={query} mt={50} mx={20} />
+  }
+
+  tracking.trackEvent({
+    action_type:
+      options.length > 0
+        ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
+        : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+    query: query,
+  })
+
+  const formattedOptions: SuggionItemOptionProps[] = formatOptions(
+    options as SearchNodeOption[]
+  )
+
+  const handleLoadMore = () => {
+    if (!relay.hasMore() || relay.isLoading()) {
+      return
+    }
+
+    setIsLoading(true)
+
+    relay.loadMore(ENTITIES_PER_SCROLL, err => {
+      if (err) {
+        console.error(err)
+      }
+
+      setIsLoading(false)
+    })
+  }
+
+  const handleOnRedirect = () => {
+    onClose()
+  }
+
+  return (
+    <>
+      {formattedOptions.map((option, index) => {
+        return (
+          <NewSuggestionItem
+            query={query}
+            option={option}
+            onRedirect={handleOnRedirect}
+            key={index}
+          />
+        )
+      })}
+
+      {relay.hasMore() && <InfiniteScrollSentinel onNext={handleLoadMore} />}
+
+      {!relay.hasMore() && (
+        <SeeFullResults query={query} onClick={handleOnRedirect} />
+      )}
+
+      {isLoading && (
+        <Flex width="100%" minHeight="60px" alignItems="center">
+          <Spinner position="relative" />
+        </Flex>
+      )}
+    </>
+  )
+}
+
+const SearchResultsListPaginationContainer = createPaginationContainer(
```

Comment:
**question**: should we support pagination for mobile view? As far as I can see by [design in Figma](https://www.figma.com/file/i5mJhlbbC4XEvvrwpImjGp/Search-Experience---Ideas-%2B-Mocks?type=design&node-id=336-44450&t=dpbI9xlBp8o3WPDo-0), instead of pagination, we should display the "See full results" button

Reply:
There is an "Advanced Options" [section](https://www.figma.com/file/i5mJhlbbC4XEvvrwpImjGp/Search-Experience---Ideas-%2B-Mocks?type=design&node-id=336-22330&t=EyiCog9nve22KiD1-0) which suggests implementing infinite scrolling. In this case it probably doesn't make sense to implement "See full results" link like you said here: https://github.com/artsy/force/pull/12410#discussion_r1195476108

I will remove "See full results" then